compiler/optimizer/OMROptimizer.cpp

```diff
@@ -2050,6 +2050,10 @@ int32_t OMR::Optimizer::performOptimization(const OptimizationStrategy *optimiza
          }
 
       delete opt;
+      // we cannot easily invalidate during IL gen since we could be peeking and we cannot destory our
+      // caller's alias sets
+      if (!isIlGenOpt())
```

Comment:
isIlGenOpt will return true in both cases - we found it in the context of peeking but isIlGenOpt will be true for peeking or during inlining. Basically, only after 'true' optimization steps that are not nested can we invalidate the alias set which is the intent behind this check.